src/library/Cardano/CoinSelection/Fee.hs

```diff
@@ -130,37 +151,64 @@ data FeeOptions i o = FeeOptions
 
     , dustThreshold
         :: DustThreshold
-        -- ^ Change addresses below the given threshold will be evicted
-        -- from the created transaction. This can be useful if you do not want
-        -- to produce change output below a certain threshold and instead,
-        -- conflate them with other change outputs.
-        --
-        -- Setting the 'dustThreshold' to @mempty@ is a very reasonable choice
-        -- if you don't know what to do with this. This would have no effect and
-        -- keep all valid coin values that are positive.
+        -- ^ The threshold to use for dust elimination. Specifying a threshold
+        -- of zero will disable dust elimination. See 'DustThreshold' for more
+        -- details.
 
     , feeBalancingPolicy
         :: FeeBalancingPolicy
         -- ^ What do to when we encounter a dangling change output.
         -- See 'FeeBalancingPolicy'
     } deriving Generic
 
--- | A /dangling/ change output is one that would be too expensive to add. This
--- can happen when a change output is small, and including it would result in a
--- higher fee than if it were not included.
+-- | A choice of fee balancing policies for use when adjusting a coin selection.
+--
+-- == Background
+--
+-- A coin selection __'s'__ is said to have a /perfectly-balanced/ fee when it
+-- satisfies the following property:
+--
+-- >>> sumInputs s = sumOutputs s + sumChange s + estimateFee s
+--
+-- Conversely, a selection is said to have an /unbalanced/ fee when it
+-- satisfies the following property:
+--
+-- >>> sumInputs s > sumOutputs s + sumChange s + estimateFee s
+--
+-- In other words, if a coin selection has an /unbalanced/ fee, the /effective/
+-- fee is greater than the minimum fee /actually required/ by the blockchain.
+--
+-- == Balanced Fees vs Minimal Fees
+--
+-- Some blockchains /require /that fees are always /perfectly-balanced/.
+--
+-- However, for blockchains that allow /unbalanced/ fees, it is sometimes
+-- possible to /save money/ by generating a coin selection with an unbalanced
+-- fee. This may seem counterintuitive at first, but consider an individual
+-- change ouput __/c/__ of value __/v/__. If the /marginal fee/ __/f/__
+-- associated with __/c/__ is greater than its value __/v/__, then we will
+-- /save money/ by __not__ including __/c/__ within 'change'.
```

Comment:
would be perfect to see it in real examples. show fee balancing policy that exemplifies it and see in code here

Reply:
> would be perfect to see it in real examples. show fee balancing policy that exemplifies it and see in code here. Without it it is still close to perfect;-)

Great idea. I'll see if I can include it in another PR!